hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java

```diff
@@ -2191,4 +2194,36 @@ private V mergeLocally(K key, V value, BiFunction<? super V, ? super V, ? extend
         }
     }
 
+    @Override
+    public void replaceAll(@Nonnull BiFunction<? super K, ? super V, ? extends V> function) {
+        checkNotNull(function, NULL_BIFUNCTION_IS_NOT_ALLOWED);
+        replaceAllInternal(function);
+    }
+
+    protected void replaceAllInternal(BiFunction<? super K, ? super V, ? extends V> function) {
+        if (SerializationUtil.isClassStaticAndSerializable(function) && !useDefaultReplaceAllOperation) {
+            try {
+                int partitionCount = getContext().getPartitionService().getPartitionCount();
+                List<Future<ClientMessage>> futures = new ArrayList<>(partitionCount);
+                Data functionAsData = toData(function);
+                for (int partitionId = 0; partitionId < partitionCount; partitionId++) {
```

Comment:
I think that you can group invocation per member and send a single request to each member instead of partition count requests which will be more efficient. You can even do this at the server side.

Reply:
@ihsandemir  idea is to invoke parallel requests from client to server. As we are already sending parallel request, I doubt it will be costly. @sancar  fyi

Reply:
@Prabusrit We discussed it with @sancar. I think we can do this: Let's have your solution in and open another issue to optimize this for `putAll`, `replaceAll`, `getAll`, etc. kind of invocations, all of which seems to be candidates for this improvement.

Reply:
sure